### ISSUE TITLE
Report bugs and the way to fix them

### ISSUE THREAD

**Issue by wyhwhy** [kxfan2002/SophiaVL-R1#2]
#### 1. "requirements.txt"
The present version of requirements.txt is incomplete. Please remember to add: "**rouge_score**", "**jiwer**". Besides, I successfully ran the code on the environment with **H800** and **cuda12.4**: 
```
vllm==0.8.5.post1 + transformers==4.51.1 + torch==2.6 + flash_attn==2.7.2.post1.
```
#### 2. Missing line in "verl/trainer/ray_trainer.py"
In Line 319, remember to add 
```
processor=all_dataset[0].processor, 
```
otherwise it would raise a missing argument error.
#### 3. Dataset format
Please remember to modify "verl/data/dataset_info.json" since the .json file names are different from [SophiaVL-R1-130k](https://huggingface.co/datasets/bunny127/SophiaVL-R1-130k/tree/main/json) on hugging face.

#### 4. "scripts/train_scripts/run_train.sh"
Remember to modify Line 7 into 
```
export REWARD_MODEL="thinking-reward-model"
```
which should be the same as **--served-model-name** given in "scripts/train_scripts/thinking_reward_model.sh".
Besides, remember to delete export OPENAI_API_KEY="" and add
```
export OPENAI_API_URL="http://localhost:8000/v1/chat/completions"
```
where 8000 (very **suggested** rather than 80 in the original script) should be the same as **--port 8000** in "scripts/train_scripts/thinking_reward_model.sh". Otherwise it would raise Client error 307, 401, or 404.

Thanks for the great work from the authors and I will try to report further issues when running the code.

**Comment by kxfan2002** [comment]
Thank you for your detailed feedback — I really appreciate it. Your report is very helpful and will definitely benefit others as well. I will fix these issues as soon as possible.

Just to clarify a couple of points:

1. The `REWARD_MODEL` name must match the `--served-model-name` argument used when launching the vllm server.

2. Normally, you do not need an API_KEY when calling vllm. The `port` depends on how you deploy vllm. In my case, I did not deploy the reward model locally, so I used --port 80 due to firewall restrictions.

Thanks again for your contribution! Let me know if you encounter anything else.

**Comment by kxfan2002** [comment]
The total number of training steps is much higher than 1500, but reaching 1500 steps already takes around 2–3 days. Training for a full epoch would be too computationally expensive for me :(  We found that training for 1500 steps already gives good results in practice.

I use 2 separate GPUs for serving the reward model. These are completely independent from the 8 GPUs used for training. If you adopt the same setup, make sure the specified `--port` is open for external access, and update the corresponding URL in `scripts/train_scripts/run_train.sh` accordingly.

**Comment by kxfan2002** [comment]
Hi, our reward model is trained using supervised fine-tuning (SFT), not reinforcement learning (RL). We use [LLaMA-Factory](https://github.com/hiyouga/LLaMA-Factory) for training. If you decide to try training the reward model with RL, I would be very curious to see how it performs :)

**Comment by wyhwhy** [comment]
> > > Hi, our reward model is trained using supervised fine-tuning (SFT), not reinforcement learning (RL). We use [LLaMA-Factory](https://github.com/hiyouga/LLaMA-Factory) for training. If you decide to try training the reward model with RL, I would be very curious to see how it performs :)
> > 
> > 
> > Hi, I am trying to train my own reward model with SFT like the following way. Is there any easy method recommended? Thanks! ![Image](https://github.com/user-attachments/assets/5a61df5b-3090-4574-ab43-fe98b389859b)
> 
> As far as I know, LLaMA-Factory provides an easy-to-use interface for implementing supervised fine-tuning (SFT). I have not tried other methods yet.

Thanks for your reply!
